pkg/querier/tripperware/queryrange/retry.go

```diff
@@ -84,3 +87,20 @@ func (r retry) Do(ctx context.Context, req tripperware.Request) (tripperware.Res
 	}
 	return nil, lastErr
 }
+
+func isRetryableResponse(resp *httpgrpc.HTTPResponse) bool {
+	if resp.Code/100 != 5 {
+		return false
+	}
+
+	// If pool exhausted, retry at query frontend might make things worse.
+	// Rely on retries at querier level only.
+	if strings.Contains(yoloString(resp.Body), pool.ErrPoolExhausted.Error()) {
+		return false
+	}
+	return true
+}
+
+func yoloString(buf []byte) string {
```

Comment:
I guess you meant to rename this func?

Reply:
That's the name :)
We have same function in other places.